File: Question_Go/11_Container_With_Most_Water.go
func maxArea(height []int) int {
    ans := 0
    left, right := 0, len(height) - 1
    for left < right {
        curH := height[left]
        if curH > height[right] {
            curH = height[right]
        }

        cur := (right - left) * curH
        if cur > ans {
            ans = cur
        }

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }
    return ans
}

File: Question_Java/11_Container_With_Most_Water.cpp
class Solution {
public
    int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int ans = 0;
        while (left < right) {
            ans = Math.max(
                ans, Math.min(height[left], height[right]) * (right - left));
            if (height[left] < height[right])
                ++left;
            else
                --right;
        }
        return ans;
    }
}